docs/en/stack/ml/nlp/ml-nlp-autoscaling.asciidoc

```diff
@@ -0,0 +1,144 @@
+Adaptive allocations must be set up independently for each deployment and {infer} endpoint.
+
+
+[discrete]
+[[optimize-use-case]]
+=== Optimize for typical use cases by using adaptive allocations
+
+You can optimize your model deplyoment for typical use cases, such as search and ingest.
+When you optimize for ingest, the throughput will be higher, which increases the number of {infer} requests that can be performed in parallel.
+When you optimize for search, the latency will be lower during search processes.
+
+* If you want to optimize for ingest, set the number of threads to `1` (`"num_threads": 1`).
+* If you want to optimize for search, set the number of threads to greater than `1`.
+Increasing the number of threads will make the search processes more performant.
+
+
+[discrete]
+[[nlp-model-adaptive-resources]]
+== Enabling autoscaling in {kib} - adaptive resources
+
+You can enable adaptive resources for your models when starting or updating the model deployment.
+Adaptive resources make it possible for {es} to scale up or down the available resources based on the load on the process.
+This can help you to manage performance and cost more easily.
+When adaptive resources are enabled, the number of vCPUs that the model deployment uses is set automatically based on the current load.
+When the load is high, the number of vCPUs that the process can use is automatically increased.
+When the load is low, the number of vCPUs that the process can use is automatically decreased.
+
+You can choose from three levels of resource usage for your trained model deployment.
+Refer to the tables in the <<auto-scaling-matrix>> section to find out the setings for the level you selected.
+
+
+[role="screenshot"]
+image::images/ml-nlp-deployment-id-elser-v2.png["ELSER deployment with adaptive resources enabled.",width=640]
+
+
+[discrete]
+[[auto-scaling-matrix]]
+== Model deployment resource matrix
+
+The used resources for trained model deployments depends on three factors:
+
+* your cluster environment (Serverless, Cloud, or on-premises)
+* the use case you optimize the model deployment (ingest or search)
+* whether adaptive resources are enabled or disabled (dynamic or static resources)
+
+If you use {es} on-premises, adaptive resources behavior is fully dynamic and highly dependent on the hardware configuration.
+The following tables show you the number of allocations, threads, and vCPUs available in Cloud when adaptive resources are enabled or disabled.
+
+
+[discrete]
+=== Deployments in Cloud optimized for ingest
+
+In case of ingest-optimized deployments, we maximize the number of model allocations.
+
+
+[discrete]
+==== Adaptive resources enabled
+
+[cols="4*", options="header"]
+|==========
+| Level  | Allocations                                  | Threads | vCPUs
+| Low    | 0 to 2 dynamically                           | 1       | 0 to 2 dynamically 
```

Comment:
Almost everywhere "if available" is added, Why not here? It also applies to the small numbers.